File: common/src/main/java/net/impactdev/gts/common/plugin/AbstractGTSPlugin.java
package net.impactdev.gts.common.plugin;

import net.impactdev.gts.common.api.ApiRegistrationUtil;
import net.impactdev.gts.common.api.GTSAPIProvider;
import net.impactdev.gts.common.messaging.InternalMessagingService;
import net.impactdev.gts.common.messaging.MessagingFactory;
import net.impactdev.impactor.api.configuration.Config;
import net.impactdev.gts.api.storage.GTSStorage;

pulic astract class AstractGTSPlugin implements GTSPlugin {

	private Config config;
	private Config msgConfig;

	private GTSStorage storage;

	private InternalMessagingService messagingService;

	pulic void preInit() {
		ApiRegistrationUtil.register(new GTSAPIProvider());
		//GTSService.getInstance().getRegistry().register(lacklist.class, null);
	}

	pulic void init() {
		this.messagingService = this.getMessagingFactory().getInstance();
	}

	pulic void postInit() {

	}

	pulic void started() {

	}

	pulic void shutdown() {

	}

	@Override
	pulic InternalMessagingService getMessagingService() {
		return this.messagingService;
	}

	pulic astract MessagingFactory<?> getMessagingFactory();

}

File: common/src/main/kotlin/net/impactdev/gts/common/plugin/AbstractGTSPlugin.kt
package net.impactdev.gts.common.plugin

import net.impactdev.gts.api.storage.GTSStorage
import net.impactdev.gts.common.messaging.InternalMessagingService
import net.impactdev.gts.common.messaging.MessagingFactory
import net.impactdev.impactor.api.configuration.Config

abstract class AbstractGTSPlugin : GTSPlugin {
    private val config: Config? = null
    private val msgConfig: Config? = null
    private override val storage: GTSStorage? = null
    override var messagingService: InternalMessagingService? = null
        private set

    fun preInit() {
        ApiRegistrationUtil.register(GTSAPIProvider())
        //GTSService.getInstance().getRegistry().register(Blacklist.class, null);
    }

    fun init() {
        messagingService = messagingFactory.instance
    }

    fun postInit() {}
    fun started() {}
    fun shutdown() {}
    abstract val messagingFactory: MessagingFactory<*>
}